opencensus/stats/view_data.py

```diff
@@ -0,0 +1,110 @@
+from opencensus.stats import measure
+from opencensus.stats import view
+from opencensus.tags import tag_value
+from opencensus.tags import tag_map
+from datetime import datetime
+
+
+class ViewData(object):
+    """View Data is the aggregated data for a particular view
+
+    :type view:
+    :param view: The view associated with this view data
+
+    :type start_time: datetime
+    :param start_time: the start time for this view data
+
+    :type end_time: datetime
+    :param end_time: the end time for this view data
+
+    :type rows: list(:class: '~opencensus.tags.tag.Tag', :class: '~opencensus.stats.aggregation_data.AggregationData')
+    :param rows: the collected value for a specific set of tags
+
+    """
+    def __init__(self, view, start_time, end_time, rows=None):
+        self._view = view
+        self._start_time = start_time
+        self._end_time = end_time
+        self._rows = rows if rows is not None else {}
+        self._tag_value_aggregation_map = {}
+        self._tag_map = {}
+
+    @property
+    def view(self):
+        """the current view in the view data"""
+        return self._view
+
+    @property
+    def start_time(self):
+        """the current start time in the view data"""
+        return self._start_time
+
+    @property
+    def end_time(self):
+        """the current end time in the view data"""
+        return self._end_time
+
+    @property
+    def rows(self):
+        """the current rows in the view data"""
+        return self._rows
+
+    @property
+    def tag_value_aggregation_map(self):
+        """the current tag value aggregation map in the view data"""
+        return self._tag_value_aggregation_map
+
+    def start(self):
+        """sets the start time for the view data"""
+        self._start_time = datetime.utcnow().isoformat() + 'Z'
+
+    def end(self):
+        """sets the end time for the view data"""
+        self._end_time = datetime.utcnow().isoformat() + 'Z'
+
+    def get_tag_map(self, context):
+        """function to return the tag map based on the context"""
+        if context.items() <= self._tag_map.items():
+            return self._tag_map
+        else:
+            tags = self._tag_map
+            for tag_key, tag_value in context.items():
+                tags[tag_key] = tag_value
+            return tags
+
+    def get_tag_values(self, tags, columns):
+        """function to get the tag values from tags and columns"""
+        tag_values = []
+        i = 0
+        while i < len(columns):
+            tag_key = columns[i]
+            if tag_key in tags:
+                tag_values.append(tags.get(tag_key))
+            else:
+                tag_values.append(None)
+            i += 1
+        return tag_values
+
+    def record(self, context, value, timestamp):
+        """records the view data against context"""
+        tag_values = self.get_tag_values(tags=self.get_tag_map(context), columns=self.view.columns)
+        for tag_value in tag_values:
```

Comment:
Same here, you should check and update entry for unique combination of `TagValue`s instead of going through each one.